backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopAnalysisCallback.scala

```diff
@@ -354,7 +354,14 @@ final class BloopAnalysisCallback(
     }
   }
 
-  override def apiPhaseCompleted(): Unit = ()
+  override def apiPhaseCompleted(): Unit = {
+    // We collect all the generated classes during `API`. When this phase completes, we can pass to
+    // the classfiles manager all the classes that were generated by scalac. This will mark them as
+    // invalidated, so that javac (if it runs after) can be notified.
+    // NOTE: When the compilation order is `JavaThenScala`, scalac will be notified about the
+    // invalidated classfiles in `CallbackGlobal`.
+    manager.generated(classToSource.keysIterator.toArray)
```

Comment:
So, together with this spot, it means that we will call `generated` three times in the case where we use java compilation, given that we have not removed the call to `generated` in `IncrementalCommon`.
  
I think this is not ideal but it's fine. When I get the time, I will move stuff around to guarantee that "`generated` is called by the incremental compiler as soon as class files are known from either of the local java compiler, the forked java compiler or the scala compiler" without repeated calls.